phpBB/phpbb/profilefields/manager.php

```diff
@@ -479,4 +495,165 @@ public function build_insert_sql_array($cp_data)
 
 		return $cp_data;
 	}
+
+	/**
+	* Disable all profile fields of a certain type
+	*
+	* This should be called when an extension which has profile field types
+	* is disabled so that all those profile fields are hidden and do not
+	* cause errors
+	*
+	* @param string $profilefield_type_name Type identifier of the profile fields
+	*/
+	public function disable_profilefields($profilefield_type_name)
+	{
+		// Get list of profile fields affected by this operation, if any
+		$profile_fields = array();
+		$sql = 'SELECT field_id, field_ident
+			FROM ' . PROFILE_FIELDS_TABLE . "
+			WHERE field_active = 1
+				AND field_type = '" . $this->db->sql_escape($profilefield_type_name) . "'";
+		$result = $this->db->sql_query($sql);
+		while ($row = $this->db->sql_fetchrow($result))
+		{
+			$profile_fields[(int) $row['field_id']] = $row['field_ident'];
+		}
+		$this->db->sql_freeresult($result);
+
+		// If no profile fields affected, then nothing to do
+		if (!sizeof($profile_fields))
+		{
+			return;
+		}
+
+		// Update the affected profile fields to "inactive"
+		$sql = 'UPDATE ' . PROFILE_FIELDS_TABLE . "
+			SET field_active = 0
+			WHERE field_active = 1
+				AND field_type = '" . $this->db->sql_escape($profilefield_type_name) . "'";
```

Comment:
you should replace this where clause by a IN (`WHERE ' . $this->db->sql_in_set('field_id', array_keys($profile_fields))`)